crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -189,6 +284,35 @@ impl TransactionsForAccount for PendingTransactionsForAccount {
         // is equal to the account nonce
         self.txs().contains_key(&previous_nonce) || ttx.signed_tx.nonce() == current_account_nonce
     }
+
+    fn has_balance_to_cover(
+        &self,
+        ttx: &TimemarkedTransaction,
+        current_account_balances: &HashMap<IbcPrefixed, u128>,
+    ) -> bool {
+        let mut remaining_account_balances = current_account_balances.clone();
```

Comment:
maybe sum all the costs within the txs and then check against the map balances so you don't have to clone